instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/SetRoutesTest.kt

```diff
@@ -147,4 +151,29 @@ class SetRoutesTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.java)
         }
         delay(TimeUnit.SECONDS.toMillis(3))
     }
+
+    @Test
+    fun routes_observer_receives_only_valid_alternatives() = sdkTest {
+        val mockRoute = RoutesProvider.dc_short_with_invalid_alternatives(activity)
+        val routes = NavigationRoute.create(
+            mockRoute.routeResponse,
+            RouteOptions.builder()
+                .coordinatesList(mockRoute.routeWaypoints)
+                .profile(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC)
+                .build(),
+            RouterOrigin.Custom()
+        )
+
+        mapboxNavigation.setNavigationRoutes(routes)
+
+        val actual = suspendCoroutine<RoutesUpdatedResult> { continuation ->
+            mapboxNavigation.registerRoutesObserver {
+                continuation.resume(it)
+            }
+        }
+        assertEquals(routes.take(2), actual.navigationRoutes)
+        assertEquals(routes.drop(2), actual.ignoredRoutes.map { it.navigationRoute })
+        assertTrue(actual.ignoredRoutes.all { it.reason == invalidRouteReason })
```

Comment:
Do we care about specific reason NN returns? 
```suggestion
        assertTrue(actual.ignoredRoutes.all { it.reason.isNotBlankOrEmpty() })
```

Reply:
Sure. Right now NN does not pass any reason. But when they do, we'd like to pass that reason to the user. I think a strict check is OK here.